packages/runtime/test/transport/identityRecoveryKits.test.ts

```diff
@@ -0,0 +1,124 @@
+
+beforeAll(async () => {
+    const runtimeServices = await serviceProvider.launch(1, { enableDatawallet: true });
+    services = runtimeServices[0];
+
+    const datawalletDisabledServices = await serviceProvider.launch(1, { enableDatawallet: false });
+    servicesWithDisabledDatawallet = datawalletDisabledServices[0];
+}, 30000);
+
+afterAll(async () => await serviceProvider.stop());
+
+afterEach(async () => {
+    const devicesController = services.transport.identityRecoveryKits["createIdentityRecoveryKitUseCase"]["devicesController"];
+    const tokenController = services.transport.identityRecoveryKits["createIdentityRecoveryKitUseCase"]["tokenController"];
+
+    const devices = await devicesController.list();
+
+    const backupDevices = devices.filter((device) => device.isBackupDevice);
+    for (const backupDevice of backupDevices) {
+        const matchingTokens = await tokenController.getTokens({
+            "cache.content.@type": "TokenContentDeviceSharedSecret",
+            "cache.content.sharedSecret.id": backupDevice.id.toString()
+        });
+
+        for (const matchingToken of matchingTokens) {
+            await tokenController.delete(matchingToken);
+        }
+
+        await devicesController.delete(backupDevice);
+    }
+});
+
+describe("Identity Recovery Kits", () => {
+    test("should create a recovery kit", async () => {
+        const response = await services.transport.identityRecoveryKits.createIdentityRecoveryKit({
+            profileName: "profileName",
+            passwordProtection: { password: "aPassword" }
+        });
+        expect(response).toBeSuccessful();
+
+        const devices = (await services.transport.devices.getDevices()).value;
+        const backupDevices = devices.filter((device) => device.isBackupDevice);
+        expect(backupDevices).toHaveLength(1);
+
+        const backupDevice = backupDevices[0];
+
+        const tokens = (await services.transport.tokens.getTokens({ ownerRestriction: OwnerRestriction.Own, query: { expiresAt: "^9999-12-31" } })).value;
+        expect(tokens).toHaveLength(1);
+
+        expect(tokens[0].content["@type"]).toBe("TokenContentDeviceSharedSecret");
+        expect(tokens[0].content.sharedSecret.id).toBe(backupDevice.id.toString());
+    });
+
+    test("should delete a recovery kit and its token when creating a consecutive recovery kit", async () => {
+        const firstToken = (await services.transport.identityRecoveryKits.createIdentityRecoveryKit({ profileName: "profileName", passwordProtection: { password: "aPassword" } }))
+            .value;
+        const firstBackupDevice = (await services.transport.devices.getDevices()).value.find((device) => device.isBackupDevice)!;
+
+        const response = await services.transport.identityRecoveryKits.createIdentityRecoveryKit({ profileName: "profileName", passwordProtection: { password: "aPassword" } });
+        expect(response).toBeSuccessful();
```

Comment:
Variable name `secondToken` instead of `response` would help to understand the test even better so that it is clear that it is referred to the second Identity recovery kit.

Reply:
`secondTokenResponse` would help as well, like you did with `getTokenResponse` and `getDeviceResponse` the lines below.

Reply:
Hm, I disagree. getTokenResponse and getDeviceResponse are assertions, response is ALWAYS the action itself in this test file.

Reply:
Fine for me, just one detail I stumbled across.